app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/generator/LinkPreviewGenerator.kt

```diff
@@ -18,13 +18,11 @@ object LinkPreviewGenerator : PreviewGenerator() {
     }
 
     private fun generatePreview(source: Path): Bitmap {
-        val zip = ZipFile(source.toFile())
-        val entries = zip.entries()
-        val imageEntry = entries
-            .asSequence()
-            .find { entry -> entry.name == IMAGE_FILE }
-            ?: error("No image inside .link file")
-
-        return BitmapFactory.decodeStream(zip.getInputStream(imageEntry))
+        val previewBytes =
+            loadLinkPreview(source.pathString) ?: error("No image inside .link file")
```

Comment:
Does `error` produce toast or write to log?

Reply:
Yes, we [write](https://github.com/ARK-Builders/ARK-Navigator/blob/4b20d31b22da9aa487e217b6b5ad7865bbf35887/app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/PlainResourcesIndex.kt#L229) to log if we failed to generate preview